src/mixins/object_origin.mixin.ts

```diff
@@ -1,345 +1,404 @@
+   * @type Number
+   * @default 0
+   */
+  top: number;
+
+  /**
+   * Left position of an object. Note that by default it's relative to object left. You can change this by setting originX={left/center/right}
+   * @type Number
+   * @default 0
+   */
+  left: number;
+
+  /**
+   * Object width
+   * @type Number
+   * @default
+   */
+  width: number;
+
+  /**
+   * Object height
+   * @type Number
+   * @default
+   */
+  height: number;
+
+  /**
+   * Object scale factor (horizontal)
+   * @type Number
+   * @default 1
+   */
+  scaleX: number;
+
+  /**
+   * Object scale factor (vertical)
+   * @type Number
+   * @default 1
+   */
+  scaleY: number;
+
+  /**
+   * Angle of skew on x axes of an object (in degrees)
+   * @type Number
+   * @default 0
+   */
+  skewX: number;
+
+  /**
+   * Angle of skew on y axes of an object (in degrees)
+   * @type Number
+   * @default 0
+   */
+  skewY: number;
+
+  /**
+   * Horizontal origin of transformation of an object (one of "left", "right", "center")
+   * See http://jsfiddle.net/1ow02gea/244/ on how originX/originY affect objects in groups
+   * @type String
+   * @default 'left'
+   */
+  originX: TOriginX;
+
+  /**
+   * Vertical origin of transformation of an object (one of "top", "bottom", "center")
+   * See http://jsfiddle.net/1ow02gea/244/ on how originX/originY affect objects in groups
+   * @type String
+   * @default 'top'
+   */
+  originY: TOriginY;
+
+  /**
+   * Angle of rotation of an object (in degrees)
+   * @type Number
+   * @default 0
+   */
+  angle: TDegree;
+
+  /**
+   * Width of a stroke used to render this object
+   * @type Number
+   * @default 1
+   */
+  strokeWidth: number;
+
+  /**
+   * When `false`, the stoke width will scale with the object.
+   * When `true`, the stroke will always match the exact pixel size entered for stroke width.
+   * this Property does not work on Text classes or drawing call that uses strokeText,fillText methods
+   * default to false
+   * @since 2.6.0
+   * @type Boolean
+   * @default false
+   * @type Boolean
+   * @default false
+   */
+  strokeUniform: boolean;
+
+  /**
+   * Object containing this object.
+   * can influence its size and position
+   */
+  group?: Group;
+
+  _originalOriginX?: TOriginX;
+
+  _originalOriginY?: TOriginY;
+
+  /**
+   * Calculate object bounding box dimensions from its properties scale, skew.
+   * @param {Object} [options]
+   * @param {Number} [options.scaleX]
+   * @param {Number} [options.scaleY]
+   * @param {Number} [options.skewX]
+   * @param {Number} [options.skewY]
+   * @private
+   * @returns {Point} dimensions
+   */
+  _getTransformedDimensions(options: any = {}): Point {
+    const dimOptions = {
+      scaleX: this.scaleX,
+      scaleY: this.scaleY,
+      skewX: this.skewX,
+      skewY: this.skewY,
+      width: this.width,
+      height: this.height,
```

Comment:
I put width and height here a while back
But now (after thinking of a fix to skewing controls + stroke projection) I understand it is wrong.
I makes no sense that a consumer of this method will provide the size.
Does it? The whole point of the method is to return the size of transform.
Passing width/height lets you "project" the bbox with a transform....
Thoughts?